tpool/tpool_structs.h

```diff
@@ -107,61 +111,65 @@ template<typename T> class cache
     return m_pos == capacity();
   }
 
+  /**
+   Wait on condition. Instrumented (wait time is recorded),
+   if timed template parameter is true.
+  */
+  void condition_wait(std::unique_lock<std::mutex>& lock)
+  {
+    if (timed)
+    {
+      auto start= std::chrono::high_resolution_clock::now();
+      m_cv.wait(lock);
+      m_wait_time+= std::chrono::high_resolution_clock::now() - start;
```

Comment:
How often is this code executed? I see that it is enabled for `io_slots::m_cache` but not anything else. This would seem to be executed only in `io_slots::acquire()` when there are no vacant slots, right?

Reply:
yes, and it is rather tricky to get this code executed, normally.  But loading large bufferpool, 10+ GB will, for the read cache the wait time will be about the same as it took bufferpool to load.  I think I did see 0.4 or so seconds for writes, but it required innodb_doublewrite=0, and innodb_write_io_threads = 2, and heavy write benchmark. This won't execute often anyway, with the only exception of unthrottled bufferpool load.